In [1]:
import pandas as pd 
import numpy as np

1. Businnes problem.
2. Planning solution.
3. What's python.
4. How to write Python's code.
5. The First Python commands.
6. Answer the businnes problem's questions.
7. Exercicies.

1. O problema de negócio.
2. Planejamento de solução.
3. O que Python?.
4. Como escrever código em python.
5. Os primeiros comandos Python.
6. Responder as questões de negócio.
7. Exercicios.

##  HOUSE ROCKET
    
    Property buying and selling company where the CEO of the company would like to maximize the company's profit by finding good deals.
    
    Empresa de compra e venda de propriedades onde o CEO da empresa gostaria de maximizar o lucro da empresa encontrando bons negocios.
    
    

## CEO's questions
   1. [How many houses are available to buy ?](#1.-how-many-houses-are-available-to-buy-?)
   2. [How many attributes do the houses have ?](#2.-how-many-attributes-do-the-houses-have-?)
   3. [Which attributes ?](#3.-which-attributes-?)
   4. [What's the most expensive house of the dataset ?](#4.-what's-the-most-expensive-house-of-the-dataset-?)
   5. [Which house has more betrooms ?](#5.-which-house-has-more-betrooms-?)
   6. [What's the total amount of bedrooms of the dataset ?](#6.-what's-the-total-amount-of-bedrooms-of-the-dataset-?)
   7. [How many houses has 2 bathrooms ?](#7.-how-many-houses-has-2-bathrooms-?)
   8. [What's the average  dataset houses price ?](#8.-what's-the-average-dataset-houses-price-?)
   9. [What's the average price of 2 bathrooms homes ?](#9.-what's-the-average-price-of-2-bathrooms-homes-?)
   10. [What's the lowest price for 3 bedrooms homes ?](#10.-What's-the-lowest-price-for-3-bedrooms-homes-?)
   11. [How many houses has more than 300 square meters of living room ?](#11.-how-many-houses-has-more-than-300-square-meters-of-living-room-?)
   12. [How many houses have more than 2 floors ?](#12.-how-many-houses-have-more-than-2-floors-?)
   13. [How many houses have a  sea view ?](#13.-how-many-houses-have-a-sea-view-?)
   14. [Of the houses overlooking the sea, how many have more than 3 bedrooms ?](#14.-Of-the-houses-overlooking-the-sea,-how-many-have-more-than-3-bedrooms-?)
   15. [Of the houses with more than 300 square meters of living room, how many have more than 2 bathrooms ?](#15.-Of-the-houses-with-more-than-300-square-meters-of-living-room,-how-many-have-more-than-2-bathrooms-?)

## As perguntas dos CEO
   1. [Quantas casas estão disponíveis para compra ?](#1.-how-many-houses-are-available-to-buy-?)
   2. [Quantos atributos as casas possuem?(n° quartos,n° garagens,m²,vista para o mar)](#2.-how-many-attributes-do-the-houses-have-?)
   3. [Quais os atributos ?](#3.-which-attributes-?)
   4. [Qual a casa mais cara do portifólio(casa com maior valor)?](#4.-what's-the-most-expensive-house-of-the-dataset-?)
   5. [Qual a casa com maior número de quartos?](#5.-which-house-has-more-betrooms-?)
   6. [Qual a soma total de quartos do conjunto de dados?](#6.-what's-the-total-amount-of-bedrooms-of-the-dataset-?)
   7. [Quantas casas possuem 2 banheiros?](#7.-how-many-houses-has-2-bathrooms-?)
   8. [Qual o preço médio de todas as casas no conjunto de dados?](#8.-what's-the-average-dataset-houses-price-?)
   9. [Qual o preço médio de casas com 2 banheiros?](#9.-what's-the-average-price-of-2-bathrooms-homes-?)
   10. [Qual o preço minimo entre as casas com 3 quartos?](#10.-What's-the-lowest-price-for-3-bedrooms-homes-?)
   11. [Quantas casas possuem mais de 300 metros quadrados na sala de estar?](#11.-how-many-houses-has-more-than-300-square-meters-of-living-room-?)
   12. [Quantas casas tem mais de 2 andares?](#12.-how-many-houses-have-more-than-2-floors-?)
   13. [Quantas casas tem vista para o mar?](#13.-how-many-houses-have-a-sea-view-?)
   14. [Das casas com vista para o mar, quantas tem 3 quartos?](#14.-Of-the-houses-overlooking-the-sea,-how-many-have-more-than-3-bedrooms-?)
   15. [Das casas com mais de 300 metros quadrados de sala de estar, quantas tem mais de 2 banheiros7](#15.-Of-the-houses-with-more-than-300-square-meters-of-living-room,-how-many-have-more-than-2-bathrooms-?)

## Planning the solution
    1. Planning the final product
        1.1 What i will deliver ?
            A: Text
        1.2 How the delivery will be ?
            A: Questions | Answers
    2. Process planning
        2.1 Were is the information ?
            A: dataset on kaggle
        2.2 How colect data ?
            A: public available
        2.3 Answer the questions
    3. Tool planning ?
            A: Python

## Planejameno da solução
    1. Planejamento do produto final
        1.1 Oque eu vou entregar ?
            R: Texto
        1.2 Como vai ser a entrega ?
            R: Perguntas | Respostas
    2. Planejamento do processo
        2.1 Onde está a informação ?
            R: dataset no kaggle
        2.2 Como coletar as informações ?
            R: diponiveis publicamente
        2.3 responder as perguntas
    3. Planejamento do ferramentas ?
            R: Python

![purple-divider](https://user-images.githubusercontent.com/7065401/52071927-c1cd7100-2562-11e9-908a-dde91ba14e59.png)
## Read the file

In [3]:
df = pd.read_csv('data/kc_house_data.csv')

In [4]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [5]:
df.shape

(21613, 21)

### Check for duplicates

In [6]:
df.duplicated().value_counts()

False    21613
dtype: int64

In [7]:
df.duplicated(subset=['id']).value_counts()

False    21436
True       177
dtype: int64

The dataset has some duplicate ids, but the other columns are different, so if we check for duplicates with "pd.duplicated()", they will only be shown if we add the parameter "subset" with the "id" column as a reference

In [8]:
df.drop_duplicates(inplace=True)

![purple-divider](https://user-images.githubusercontent.com/7065401/52071927-c1cd7100-2562-11e9-908a-dde91ba14e59.png)
### 1. how many houses are available to buy ?

In [9]:
len(df['id'].unique())

21436

A: There are 21436 houses to buy

### 2. how many attributes do the houses have ?

In [10]:
len(df.columns)

21

A: On this dataset we have 19 attributes for each house, because we don't take in account "id" and "date"

### 3. which attributes ?

In [11]:
print("A:")
df.columns.tolist()[2:]

A:


['price',
 'bedrooms',
 'bathrooms',
 'sqft_living',
 'sqft_lot',
 'floors',
 'waterfront',
 'view',
 'condition',
 'grade',
 'sqft_above',
 'sqft_basement',
 'yr_built',
 'yr_renovated',
 'zipcode',
 'lat',
 'long',
 'sqft_living15',
 'sqft_lot15']

### 4. what's the most expensive house of the dataset ?

In [12]:
df[df['price'] == df['price'].max()]

id             date      price  bedrooms  bathrooms  \
7252  6762700020  20141013T000000  7700000.0         6        8.0   

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
7252        12050     27600     2.5           0     3  ...     13        8570   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
7252           3480      1910          1987    98102  47.6298 -122.323   

      sqft_living15  sqft_lot15  
7252           3940        8800  

[1 rows x 21 columns]

A: the most expensive house costs 7.700.000 U$ and has the id: 6762700020

### 5. which house has more betrooms ?

In [13]:
df[df['bedrooms']==df['bedrooms'].max()]

id             date     price  bedrooms  bathrooms  \
15870  2402100895  20140625T000000  640000.0        33       1.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15870         1620      6000     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15870        1040            580      1947             0    98103  47.6878   

          long  sqft_living15  sqft_lot15  
15870 -122.331           1330        4700  

[1 rows x 21 columns]

A: the house which more bedrooms has 33 bedrooms in total and the id: 2402100895

### 6. what's the total amount of bedrooms of the dataset ?

In [14]:
df['bedrooms'].sum()

72854

A: the dataset has 72854 bedrooms

### 7. how many houses has 2 bathrooms ?

In [15]:
len(df[df['bathrooms']==2]['id'])

1930

A: 1930 houses has 2 bathrooms

### 8. what's the average dataset houses price ?

In [16]:
round(df['price'].mean(),3)

540088.142

A: the average price's houses are 540,088.142 U$

### 9. what's the average price of 2 bathrooms homes ?

In [17]:
round(df[df['bathrooms']==2]['price'].mean(),3)

457889.719

A: the average price of 2 bathrooms homes are 457,889.719 U$

###  10. What's the lowest price for 3 bedrooms homes ?

In [18]:
df[df['bedrooms']==3]['price'].min()

82000.0

A: the lowest price for 3 bedrroms homes is 82,000.00 U$

### 11. how many houses has more than 300 square meters of living room ?

In [19]:
df['m2'] = df['sqft_living']*0.09290304
len(df[df['m2']>300])

2258

A: 21612 houses has more than 300 m² of living room

### 12. how many houses have more than 2 floors ?

In [20]:
len(df[df['floors']>2])

782

A: 782 houses has more than 2 floors

### 13. how many houses have a sea view ?

In [21]:
len(df[df['waterfront']==True])

163

A: 163 houses has sea view

###  14. Of the houses overlooking the sea, how many have more than 3 bedrooms ?

In [22]:
len(df[(df['waterfront']==True)&(df['bedrooms']>3)])

63

A: 63 houses have 3 bedrooms ans sea view

### 15. Of the houses with more than 300 square meters of living room, how many have more than 2 bathrooms ?

In [23]:
len(df[(df['m2']>300)&(df['bathrooms']>2)])

2201

A: 2201 houses have moren than 300 square meters of living room and more than 2 bathrooms